### ISSUE TITLE
Some spans are being skipped by spacy-huggingface-pipelines, resulting in poor anonymisation

### ISSUE THREAD

**Issue by aayushisanghi** [microsoft/presidio#1262]
Hi! I've been working on a transformer-based Presidio pipeline, and I noticed it was performing rather poorly. Upon inspecting the logs, I found this particular warning:
`UserWarning: Skipping annotation, {'entity_group':'PASSWORD', 'score': 0.25415105, 'word': '##bmh78', 'start': 157,
"end': 1623} is overlapping or can't be aligned for doc 'Standardized tests will be...'`

The root cause of this issue is [this line](https://github.com/explosion/spacy-huggingface-pipelines/blob/main/spacy_huggingface_pipelines/token_classification.py#L129) in the spacy-huggingface-pipelines package. I know this isn't directly Presidio related, but is there any configuration change I can make in Presidio, to prevent these spans from being skipped? Or is there another issue I'm not seeing here?

I followed the online tutorial, and am using a publicly available dataset to test different models. I'm not sure why this could be happening.

Any help to debug this will be super helpful, thanks!

**Comment by omri374** [comment]
@aayushisanghi, as @VMD7 mentioned, a reproducible example would definitely help. Thanks!

**Comment by omri374** [comment]
@aayushisanghi, we'd be very interested to know more about this issue especially as the result is poor anonymization. Any feedback would be valuable.

**Comment by thomas-moulin** [comment]
Hello! I am having the same issue. 
I am using a specific transformer model to be able to detect PII on french text. 
With this approach it seems to be able to detect the PII but for some reason it only output it in a warning log ... 


![Capture d’écran 2024-02-13 à 11 41 55](https://github.com/microsoft/presidio/assets/113366643/06262ffb-54c5-4654-a9ac-88f86b85aa9d)


Here is my code:

`import transformers
from huggingface_hub import snapshot_download
from transformers import AutoTokenizer, AutoModelForTokenClassification
from presidio_analyzer import AnalyzerEngine, RecognizerRegistry
from presidio_analyzer.nlp_engine import NlpEngineProvider


transformers_model = "Jean-Baptiste/camembert-ner-with-dates"
snapshot_download(repo_id=transformers_model)

AutoTokenizer.from_pretrained(transformers_model)
AutoModelForTokenClassification.from_pretrained(transformers_model)

conf_file = "/Users/thomasmoulin/Downloads/config_presidio_fr_transformer.yml"

provider = NlpEngineProvider(conf_file=conf_file)
nlp_engine = provider.create_engine()

analyzer = AnalyzerEngine(
    nlp_engine=nlp_engine, 
    supported_languages=["fr"]
)

result = analyzer.analyze(text="Je m'appelle Thomas Moulin", language="fr")
`

**Comment by omri374** [comment]
Thanks @thomas-moulin. Would you mind sharing your conf_file? or is it standard?

**Comment by thomas-moulin** [comment]
Thanks for the quick reply ! 

sure ! 

`nlp_engine_name: transformers
models:
  -
    lang_code: fr
    model_name:
      spacy: fr_core_news_sm
      transformers: Jean-Baptiste/camembert-ner-with-dates

ner_model_configuration:
  labels_to_ignore:
  - O
  aggregation_strategy: simple # "simple", "first", "average", "max"
  stride: 16
  alignment_mode: strict # "strict", "contract", "expand"
  model_to_presidio_entity_mapping:
    PER: PERSON
    LOC: LOCATION

  low_confidence_score_multiplier: 0.4`

**Comment by omri374** [comment]
@thomas-moulin is warning the only thing the gets outputted? I tried to reproduce this and got `[type: PERSON, start: 13, end: 26, score: 0.992917537689209]`

**Comment by thomas-moulin** [comment]
Yes in my case I only have the warning that gets outputted. the result variable is an empty list

**Comment by thomas-moulin** [comment]
![Capture d’écran 2024-02-13 à 15 54 18](https://github.com/microsoft/presidio/assets/113366643/dec85ef7-567c-4d8d-b803-818bd7470553)

**Comment by thomas-moulin** [comment]
yes @omri374 it works! 
Thank you very much for your help :)

**Comment by omri374** [comment]
Great. Leaving the issue open as there still could be corner cases where there's a wrong output.

**Comment by thomas-moulin** [comment]
Yes ! On longer inputs (OCR on french resume) I still have some warnings but not as many as before

**Comment by omri374** [comment]
Warnings are inevitable (it's part of spacy-huggingface-pipelines) but I'd be interested to see if there are missing predictions.

**Comment by fml09** [comment]
Hello @omri374, @VMD7 . I am currently experiencing the same issue. Below is the code that can reproduce the problem.
```py
from presidio_analyzer import AnalyzerEngine, RecognizerResult
from presidio_analyzer.nlp_engine import NerModelConfiguration, TransformersNlpEngine

model_config = [
    {
        "lang_code": "en",
        "model_name": {
            "spacy": "en_core_web_sm",
            "transformers": "lakshyakh93/deberta_finetuned_pii",
        },
    }
]

mapping = dict(
    USERNAME="USERNAME",
    EMAIL="EMAIL",
    KEY="KEY",
    PASSWORD="PASSWORD",
    IP_ADDRESS="IP_ADDRESS",
    FIRSTNAME="FIRSTNAME",
    LASTNAME="LASTNAME",
    MIDDLENAME="MIDDLENAME",
    IPV4="IP_ADDRESS",
    IPV6="IP_ADDRESS",
    IP="IP_ADDRESS",
    PHONE_NUMBER="PHONE_NUMBER",
    SSN="SSN",
    ACCOUNTNUMBER="ACCOUNTNUMBER",
    CREDITCARDNUMBER="CREDITCARDNUMBER",
    CREDITCARDISSUER="CREDITCARDISSUER",
    CREDITCARDCVV="CREDITCARDCVV",
)
ner_model_configuration = NerModelConfiguration(
    model_to_presidio_entity_mapping=mapping,
)
nlp_engine = TransformersNlpEngine(models=model_config, ner_model_configuration=ner_model_configuration)
engine = AnalyzerEngine(
    nlp_engine=nlp_engine,
    supported_languages=[
        "en",
    ],
)


print(
    engine.analyze(
        "My name is Clara and I live in Berkeley. this is my ip address : 175.5.0.1. this is my password: sad$f-j?ss11FF. credit card is 1231-1231-1451-2134",
        language="en",
    )
)

```

```
error: 
/spacy_huggingface_pipelines/token_classification.py:129: UserWarning: Skipping annotation, {'entity_group': 'CREDITCARDNUMBER', 'score': 0.9902746, 'word': '31-1231-1451-2134', 'start': 130, 'end': 147} is overlapping or can't be aligned for doc 'My name is Clara and I live in Berkeley. this is my ip address : 175.5.0.1. this is my password: sad...'
  warnings.warn(
```
It seems like there might be an issue with using spacy and transformers together.

**Comment by omri374** [comment]
@fml09 do you experience skipped entities or just warnings?

**Comment by fml09** [comment]
@omri374
Both of them. It skips entities and makes warning messages as well.

Result: 
```
[type: PASSWORD, start: 97, end: 111, score: 0.9996715188026428, type: CREDITCARDNUMBER, start: 128, end: 132, score: 0.9856985211372375, type: IP_ADDRESS, start: 65, end: 74, score: 0.95, type: FIRSTNAME, start: 11, end: 16, score: 0.9181937575340271, type: IN_PAN, start: 101, end: 111, score: 0.05]
```

**Comment by fml09** [comment]
@omri374 any news?

**Comment by omri374** [comment]
Looking into this. If we can't find a resolution, we will likely remove the dependency on spacy-huggingface-pipelines and call transformers directly.

**Comment by omri374** [comment]
For your specific case, please try changing the aggregation strategy to `max`:
```py
ner_model_configuration = NerModelConfiguration(
    model_to_presidio_entity_mapping=mapping,
    aggregation_strategy="max"
)
```
It will result in the credit card number to be fully identified.
